test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,386 @@
+        assert.isDefined(plot.size(), "size() is defined");
+        assert.isDefined(plot.symbol(), "symbol() is defined");
+
+        svg.remove();
+      });
     });
 
-    it("can retrieve entities in a certain range", () => {
-      let svg = TestMethods.generateSVG(400, 400);
-      let xScale = new Plottable.Scales.Linear();
-      let yScale = new Plottable.Scales.Linear();
+    describe("the Accessors properly access data, index and Dataset", () => {
+      const SVG_WIDTH = 400;
+      const SVG_HEIGHT = 400;
+      let svg: d3.Selection<void>;
+      let plot: Plottable.Plots.Scatter<number, number>;
+      let dataset: Plottable.Dataset;
+      let data: any[];
 
-      let dataset = new Plottable.Dataset([{x: 0, y: 0}, {x: 1, y: 1}]);
-      let dataset2 = new Plottable.Dataset([{x: 1, y: 2}, {x: 3, y: 4}]);
-      let plot = new Plottable.Plots.Scatter();
-      plot.x((d: any) => d.x, xScale)
-          .y((d: any) => d.y, yScale)
-          .addDataset(dataset)
-          .addDataset(dataset2);
-      plot.renderTo(svg);
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        let xScale = new Plottable.Scales.Linear().domain([0, SVG_WIDTH]);
+        let yScale = new Plottable.Scales.Linear().domain([SVG_HEIGHT, 0]);
+        let metadata = { foo: 10, bar: 20 };
+        let xAccessor = (d: any, i: number, dataset: Plottable.Dataset) => d.x + i * dataset.metadata().foo;
+        let yAccessor = (d: any, i: number, dataset: Plottable.Dataset) => dataset.metadata().bar;
+        data = [{ x: 0, y: 0 }, { x: 1, y: 1 }];
+        dataset = new Plottable.Dataset(data, metadata);
+        plot = new Plottable.Plots.Scatter<number, number>();
+        plot.x(xAccessor, xScale);
+        plot.y(yAccessor, yScale);
+        plot.addDataset(dataset);
+      });
 
-      let entities = plot.entitiesIn({ min: xScale.scale(1), max: xScale.scale(1) },
-                                     { min: yScale.scale(1), max: yScale.scale(1) });
+      it("renders symbols correctly", () => {
+        plot.renderTo(svg);
+        let symbols = plot.selections();
+        assert.strictEqual(symbols.size(), 2, "exactly 2 symbols are rendered");
 
-      assert.lengthOf(entities, 1, "only one entity has been retrieved");
-      assert.deepEqual(entities[0].datum, {x: 1, y: 1}, "correct datum has been retrieved");
+        let c1 = d3.select(symbols[0][0]);
+        let c2 = d3.select(symbols[0][1]);
+        let c1Position = d3.transform(c1.attr("transform")).translate;
+        let c2Position = d3.transform(c2.attr("transform")).translate;
+        assert.closeTo(c1Position[0], 0, 0.01, "first symbol cx is correct");
+        assert.closeTo(c1Position[1], 20, 0.01, "first symbol cy is correct");
+        assert.closeTo(c2Position[0], 11, 0.01, "second symbol cx is correct");
+        assert.closeTo(c2Position[1], 20, 0.01, "second symbol cy is correct");
```

Comment:
If these values are easy to compute let's try to compute them instead of magically dumping them? 

Please comment as we have not reached a sound conclusion regarding this best practice.